cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -576,100 +400,26 @@ func (m *gceManagerImpl) deleteNodePool(toBeRemoved *Mig) error {
 		return fmt.Errorf("only autoprovisioned node pools can be deleted")
 	}
 	// TODO: handle multi-zonal node pools.
-	deleteOp, err := m.gkeBetaService.Projects.Zones.Clusters.NodePools.Delete(m.projectId, m.location, m.clusterName,
-		toBeRemoved.nodePoolName).Do()
-	if err != nil {
-		return err
-	}
-	err = m.waitForGkeOp(deleteOp)
+	err := m.GkeService.DeleteNodePool(toBeRemoved.nodePoolName)
 	if err != nil {
 		return err
 	}
-	return m.fetchAllNodePools()
+	return m.refreshNodePools()
 }
 
 func (m *gceManagerImpl) createNodePool(mig *Mig) error {
 	m.assertGKENAP()
 
-	// TODO: handle preemptible
-	// TODO: handle SSDs
-
-	accelerators := []*gke_beta.AcceleratorConfig{}
-
-	if gpuRequest, found := mig.spec.extraResources[gpu.ResourceNvidiaGPU]; found {
-		gpuType, found := mig.spec.labels[gpu.GPULabel]
-		if !found {
-			return fmt.Errorf("failed to create node pool %v with gpu request of unspecified type", mig.nodePoolName)
-		}
-		gpuConfig := &gke_beta.AcceleratorConfig{
-			AcceleratorType:  gpuType,
-			AcceleratorCount: gpuRequest.Value(),
-		}
-		accelerators = append(accelerators, gpuConfig)
-
-	}
-
-	taints := []*gke_beta.NodeTaint{}
-	for _, taint := range mig.spec.taints {
-		if taint.Key == gpu.ResourceNvidiaGPU {
-			continue
-		}
-		effect, found := taintEffectsMap[taint.Effect]
-		if !found {
-			effect = "EFFECT_UNSPECIFIED"
-		}
-		taint := &gke_beta.NodeTaint{
-			Effect: effect,
-			Key:    taint.Key,
-			Value:  taint.Value,
-		}
-		taints = append(taints, taint)
-	}
-
-	labels := make(map[string]string)
-	for k, v := range mig.spec.labels {
-		if k != gpu.GPULabel {
-			labels[k] = v
-		}
-	}
-
-	config := gke_beta.NodeConfig{
-		MachineType:  mig.spec.machineType,
-		OauthScopes:  defaultOAuthScopes,
-		Labels:       labels,
-		Accelerators: accelerators,
-		Taints:       taints,
-	}
-
-	autoscaling := gke_beta.NodePoolAutoscaling{
-		Enabled:         true,
-		MinNodeCount:    napMinNodes,
-		MaxNodeCount:    napMaxNodes,
-		Autoprovisioned: true,
-	}
-
-	createRequest := gke_beta.CreateNodePoolRequest{
-		NodePool: &gke_beta.NodePool{
-			Name:             mig.nodePoolName,
-			InitialNodeCount: 0,
-			Config:           &config,
-			Autoscaling:      &autoscaling,
-		},
-	}
-
-	createOp, err := m.gkeBetaService.Projects.Zones.Clusters.NodePools.Create(m.projectId, m.location, m.clusterName,
-		&createRequest).Do()
-	if err != nil {
-		return err
-	}
-	err = m.waitForGkeOp(createOp)
+	err := m.GkeService.CreateNodePool(mig)
 	if err != nil {
 		return err
 	}
-	err = m.fetchAllNodePools()
+	fmt.Println("Create node pool succeeded")
```

Comment:
No println :)

Reply:
leftover debug printlns, sorry :) should be gone now